### PR TITLE
Add overlay styles and scroll lock to Modal

### PR DESCRIPTION
**Pull request by macfarlandian**
## Description of the change

In Spotlight I extended the `Modal` component with some features that I expect would be universally applicable: 

1. A translucent color and blur effect on the overlay when it's open
2. Preventing the page behind the modal from scrolling while it's open
3. Some styles to make the size of the modal more responsive to small screens and scroll overflowing content

This brings all three of those features into the Design System, and bumps the package version accordingly.

(The scroll locking behavior, unfortunately, does not work in Storybook because it doesn't seem to work in an `iframe`; however, I verified the behavior by installing the package locally in another application).

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

n/a

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Review by nbhargava — COMMENTED** [review]
I'm not sure if this is my computer or what, but the animation in Storybook for the modal is extremely slow and jerky when I'm using Chrome. When I switch back to `main`, it's relatively smooth, and on Safari it's smooth regardless.

Do you know if it's the fault of the blur + textured background vs. something else?

https://user-images.githubusercontent.com/993682/117547577-4b9bfc80-affe-11eb-80bf-0b89c09b2070.mov

File: packages/design-system/package.json [hunk 1]
```diff
@@ -22,6 +22,8 @@
   "dependencies": {
     "@fontsource/libre-baskerville": "^4.1.0",
     "@fontsource/libre-franklin": "^4.1.0",
+    "@types/body-scroll-lock": "^2.6.1",
```
[the remark this answers is not in the document]
Comment (macfarlandian) [reply packages/design-system/package.json:25]: yeah i guess it should ... for some reason I had gotten the idea that people installing the library would want these types, but looking at the exports again it doesn't actually seem like that's the case

**Review by macfarlandian — COMMENTED** [review]
@nbhargava interesting ... i saw that too but also thought it might be my overworked machine. the safari comparison suggests that may not be the case, though. I have had performance issues with this effect before and thought that I might have resolved them, but I'll take another look at it

**Comment by macfarlandian** [comment]
@nbhargava I haven't been able to really replicate the performance issue this morning, but I did make a change that only animates the `opacity` property rather than the color and the filter, which I would expect to perform better. Curious whether you see a difference on your end?

**Comment by macfarlandian** [comment]
@nbhargava do you mean on/off in the sense that if it's off, the overlay is completely unstyled? or just the animated transition is toggled? 

I think it's probably doable but it seems a little weird to me. Are we in a place where we really have different performance requirements per product? Personally I would rather we actually get this feature working to our satisfaction in all the browsers we care about or change it if we can't accomplish that.

**Comment by nbhargava** [comment]
I meant the overlay is unstyled i.e. no blur.

I don't think we want different performance requirements per product. I guess I was optimizing for let's get it in and iterate without necessarily shipping weird behaviors downstream.

I don't know that I have the right nose to determine whether this is smooth enough™ for our standards, but I'm happy to approve if you all feel like it is!

**Review by nbhargava — APPROVED** [review]
If it's already in spotlight, let's 🚢 it.

**Pull request merged**